added setPermissionMode to create files/folders with specific mode ala 0777

aldas opened the pull request:
* added `setPermissionMode` to create files/folders with specific mode ala 0777
* fixed some of the warnings that Idea/phpstorm inspectors gave
* fixed problem with path when extracting files - strpos'sing path to find directory part is not that clever as directory separator is different on windows vs linux
* `extractTo` now honors folder set with `folder()`

Chumper commented:
Wouldn't it be better to always assume 755 unless the developer passes a second parameter to `folder`?

Ot are you also setting the `permissionParameter` on files?

aldas commented:
You are probably right. Got a bit too enthusiastic/carried away with it. I'll close this PR and make another for  

* fixed some of the warnings that Idea/phpstorm inspectors gave
* fixed problem with path when extracting files - strpos'sing path to find directory part is not that clever as directory separator is different on windows vs linux
* extractTo now honors folder set with folder()
